### DISCUSSION TITLE
non-expert install instructions

### DISCUSSION THREAD

**Discussion by CouldBeThis** [resurrecting-open-source-projects/scrot#187] [Q&A] [answered]
Hello, 

Because I am using an older distro, only `scrot` 0.8 is available via `apt` repo. Therefor I was hoping to install from here to get the shiny updated version. :) 

I downloaded release `1.7` but couldn't see what to do with it. Poking around in the directory there was only one other thing that looked as though it could be useful:

```bash
$ ./install-sh 
./install-sh: no input file specified.
```

I tried the readme instructions in case they were meant to be applied, but I do not think that's correct. 

```bash
$ ./autogen.sh
zsh: no such file or directory: ./autogen.sh
```
and I tried skipping to the second step in case for some reason that would help

```bash
$ make install
make: *** No rule to make target 'install'.  Stop.
```

So I cloned the repo and attempted to follow the compile instructions from there. I solved one dependency in issue while `./configure`ing but then I couldn't solve the next one:

```
configure: error: BSD functions not found, libbsd is required
```

I searched in `apt` and found that there is a library called `libbsd0` already installed. If it is equivalent then maybe it can't be seen due to slightly different name, which isn't something I know how to fix. I saw in the readme that maybe this can be circumvented, which got me to:

```bash
$ ./configure --without-libbsd
configure: error: BSD functions not found and --without-libbsd was used
make: *** No targets specified and no makefile found.  Stop.
```

This I can't interpret, but anyway I have no particular desire to compile anything if there is a simpler method available. If the release has an installer maybe it would be helpful to include specific instructions on how to access it. 

Thank you for taking such good care of old projects.

**Comment by daltomi** [comment]
Hi,
> ```shell
> $ ./install-sh 
> ./install-sh: no input file specified.
> ```

See [Building]( https://github.com/resurrecting-open-source-projects/scrot#building) instructions, it says nothing about you having to run that script.


> ```shell
> $ ./autogen.sh
> zsh: no such file or directory: ./autogen.sh
> ```
You must run the script in the same directory it is in.


> I searched in `apt` and found that there is a library called `libbsd0` already installed


The installation of the development libraries depends a lot on what distribution you are using.
As you indicate `apt` I estimate that you should install the packages with the postfix `-dev`, ex: `libbsd-dev`, and the same for the others. You'll notice that some other package needs to be installed when `./configure` fails.

If you still have problems send us the `config.log` file (for your privacy you can edit it to **hide your username**)

**Comment by guijan** [comment]
I think the issue here is that compilation instructions for release tarballs are missing.

**Comment by eribertomota** [comment]
I noticed that bootstrapped tarball (generated from `make dist`) doesn't have autogen.sh, CONTRIBUTING.md and TODO.md.

I think it is a bit hard for a newbie to follow BUILDING instructions without find autogen.sh. For advanced users and other developers, I think that CONTRIBUTING.md and TODO.md are very important. I would like to add the last two files to dist_doc_DATA and autogen.sh to EXTRA_DIST, all in Makefile.am, and release the version 1.8.1 (with a new bootstrapped tarball).

What you think about this @daltomi?

**Reply by guijan** [reply]
README.md just needs to say autogen.sh isn't used for release tarballs.

**Reply by daltomi** [reply, marked as answer]
Hello everyone.

@eribertomota  I agree, those files can go in the tarball.

I also agree with @guijan. We should indicate that to compile this project from a tarball the normal user can use the `./configure` script
The `autogen.sh` script is for developers who want to modify scrot, as are the dependencies: `autoconf`, `autoconf-archive` and `automake`.

Possible new description, I took it from the MOC project i'm lazy now :P

```readme.md
Generic installation instruction:

In short, if you are building from an GIT checkout of scrot (but not if you
are building from a downloaded tarball) then you will first need to run:

	./autogen.sh

and then proceed as shown below for a tarball.  (If you are using the
tarball but have applied additional patches then you may also need to run
./autogen.sh.)

To build scrot from a downloaded tarball just type:

	./configure
	make

And as root:

	make install
```
Something like that I suggest.

**Comment by eribertomota** [comment]
I need a revision and an approval for the PR #210.

I added a "TIP" because there are some tarballs to download and a newbie will not able to discern between them.
